mlflow/prophet.py

```diff
@@ -0,0 +1,296 @@
+                                      registered model if one with the given name does not exist.
+        :param signature: :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                          describes model input and output
+                          :py:class:`Schema <mlflow.types.Schema>`.
+                          The model signature can be :py:func:`inferred
+                          <mlflow.models.infer_signature>` from datasets with valid model input
+                          (e.g. the training dataset with target column omitted) and valid model
+                          output (e.g. model predictions generated on the training dataset),
+                          for example:
+
+                          .. code-block:: python
+
+                            from mlflow.models.signature import infer_signature
+
+                            model = Prophet().fit(df)
+                            train = model.history
+                            predictions = model.predict(model.make_future_dataframe(30))
+                            signature = infer_signature(train, predictions)
+
+        :param input_example: Input example provides one or several instances of valid
+                              model input. The example can be used as a hint of what data to
+                              feed the model. The given example will be converted to a
+                              Pandas DataFrame and then serialized to json using the
+                              Pandas split-oriented format. Bytes are base64-encoded.
+
+        :param await_registration_for: Number of seconds to wait for the model version
+                            to finish being created and is in ``READY`` status.
+                            By default, the function waits for five minutes.
+                            Specify 0 or None to skip waiting.
+        :param pip_requirements: {{ pip_requirements }}
+        :param extra_pip_requirements: {{ extra_pip_requirements }}
+        :param kwargs: kwargs to pass to `Prophet.save_model`_ method.
+        """
+    Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.prophet,
+        registered_model_name=registered_model_name,
+        pr_model=pr_model,
+        conda_env=conda_env,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+        **kwargs,
+    )
+
+
+def _save_model(model, path):
+
+    from prophet.serialize import model_to_json
+
+    model_ser = model_to_json(model)
+    with open(path, "w") as f:
+        yaml.safe_dump(model_ser, stream=f, default_flow_style=False)
+
+
+def _load_model(path):
+
+    from prophet.serialize import model_from_json
+
+    with open(path, "r") as f:
+        model = yaml.safe_load(f)
```

Comment:
Interesting that we're using the `yaml` module here to write / read the serialized model contents produced by `model_to_json`. Is there a reason for leveraging `yaml` rather than `json.dumps / loads`? If not, can we swap in `json`?

Reply:
Definitely can! Changed.